$MirOS: ports/misc/mc/patches/patch-lib_mc_lib,v 1.2 2005/06/07 08:13:46 tg Exp $
--- lib/mc.lib.orig	Sat Aug 17 01:16:03 2002
+++ lib/mc.lib	Tue Jun  7 08:10:17 2005
@@ -1,8 +1,28 @@
+# $MirOS: ports/misc/mc/patches/patch-lib_mc_lib,v 1.2 2005/06/07 08:13:46 tg Exp $
+#-
+# Copyright (c) 2003, 2004, 2005
+#	Thorsten "mirabile" Glaser <tg@66h.42h.de>
+#
+# Licensee is hereby permitted to deal in this work without restric-
+# tion, including unlimited rights to use, publicly perform, modify,
+# merge, distribute, sell, give away or sublicence, provided all co-
+# pyright notices above, these terms and the disclaimer are retained
+# in all redistributions or reproduced in accompanying documentation
+# or other materials provided with binary redistributions.
+#
+# Licensor offers the work "AS IS" and WITHOUT WARRANTY of any kind,
+# express, or implied, to the maximum extent permitted by applicable
+# law, without malicious intent or gross negligence; in no event may
+# licensor, an author or contributor be held liable for any indirect
+# or other damage, or direct damage except proven a consequence of a
+# direct error of said person and intended use of this work, loss or
+# other issues arising in any way out of its use, even if advised of
+# the possibility of such damage or existence of a nontrivial bug.
+
 [Special dirs]
 list=/afs,/coda,/:,/...,/net,/#smb
 
 [terminal:console]
-insert=\e[2~
 f11=\e[23~
 f12=\e[24~
 f13=\e[25~
@@ -13,25 +32,25 @@ f17=\e[31~
 f18=\e[32~
 f19=\e[33~
 f20=\e[34~
-kpleft=\eOt
-kpright=\eOv
-kpup=\eOx
+insert=\e[2~
+kpasterix=\eOR
+kpdelete=\eOn
 kpdown=\eOr
-kphome=\eOw
 kpend=\eOq
+kpenter=\eOM
+kphome=\eOw
+kpinsert=\eOp
+kpleft=\eOt
+kpminus=\eOS
 kpnpage=\eOs
-kpppage=\eOy
+kpnumlock=\eOP
 kpplus=\eOl
-kpminus=\eOS
-kpasterix=\eOR
-kpinsert=\eOp
-kpdelete=\eOn
-kpenter=\eOM
+kpppage=\eOy
+kpright=\eOv
 kpslash=\eOQ
-kpnumlock=\eOP
+kpup=\eOx
 
 [terminal:linux]
-insert=\e[2~
 f11=\e[23~
 f12=\e[24~
 f13=\e[25~
@@ -42,25 +61,30 @@ f17=\e[31~
 f18=\e[32~
 f19=\e[33~
 f20=\e[34~
-kpleft=\eOt
-kpright=\eOv
-kpup=\eOx
+insert=\e[2~
+kpasterix=\eOR
+kpdelete=\eOn
 kpdown=\eOr
-kphome=\eOw
 kpend=\eOq
+kpenter=\eOM
+kphome=\eOw
+kpinsert=\eOp
+kpleft=\eOt
+kpminus=\eOS
 kpnpage=\eOs
-kpppage=\eOy
+kpnumlock=\eOP
 kpplus=\eOl
-kpminus=\eOS
-kpasterix=\eOR
-kpinsert=\eOp
-kpdelete=\eOn
-kpenter=\eOM
+kpppage=\eOy
+kpright=\eOv
 kpslash=\eOQ
-kpnumlock=\eOP
+kpup=\eOx
 
 [terminal:xterm]
-insert=\e[2~
+bs=
+delete=\e[3~
+end=\eOF
+end=\e[4~
+end=\e[8~
 f11=\e[23~
 f12=\e[24~
 f13=\e[25~
@@ -71,25 +95,21 @@ f17=\e[31~
 f18=\e[32~
 f19=\e[33~
 f20=\e[34~
-kpleft=\eOt
-kpright=\eOv
-kpup=\eOx
+home=\eOH
+home=\e[1~
+home=\e[7~
+insert=\e[2~
+kpasterix=\eOj
 kpdown=\eOr
-kphome=\eOw
 kpend=\eOq
+kphome=\eOw
+kpleft=\eOt
+kpminus=\eOm
 kpnpage=\eOs
-kpppage=\eOy
 kpplus=\eOk
-kpminus=\eOm
-kpasterix=\eOj
-delete=\e[3~
-bs=
-home=\e[1~
-home=\e[7~
-end=\eOF
-end=\e[4~
-end=\e[8~
-home=\eOH
+kpppage=\eOy
+kpright=\eOv
+kpup=\eOx
 pgdn=\e[6~
 pgup=\e[5~
 
@@ -106,7 +126,11 @@ f19=\e[20;2~
 f20=\e[21;2~
 
 [terminal:xterm-color]
-insert=\e[2~
+bs=
+delete=\e[3~
+end=\eOF
+end=\e[4~
+end=\e[8~
 f11=\e[23~
 f12=\e[24~
 f13=\e[25~
@@ -117,29 +141,27 @@ f17=\e[31~
 f18=\e[32~
 f19=\e[33~
 f20=\e[34~
-kpleft=\eOt
-kpright=\eOv
-kpup=\eOx
+home=\eOH
+home=\e[1~
+home=\e[7~
+insert=\e[2~
+kpasterix=\eOj
 kpdown=\eOr
-kphome=\eOw
 kpend=\eOq
+kphome=\eOw
+kpleft=\eOt
+kpminus=\eOm
 kpnpage=\eOs
-kpppage=\eOy
 kpplus=\eOk
-kpminus=\eOm
-kpasterix=\eOj
-delete=\e[3~
-bs=
-home=\e[1~
-home=\e[7~
-home=\eOH
-end=\e[4~
-end=\e[8~
-end=\eOF
+kpppage=\eOy
+kpright=\eOv
+kpup=\eOx
 pgdn=\e[6~
 pgup=\e[5~
 
 [terminal:ibmpc3]
+bs=^h
+end=\e[F
 f11=\e[Y
 f12=\e[Z
 f13=\e[a
@@ -150,8 +172,140 @@ f17=\e[e
 f18=\e[f
 f19=\e[g
 f20=\e[h
-bs=^h
-end=\e[F
-kpplus=\e[+
-kpminus=\e[-
 kpasterix=\e[*
+kpminus=\e[-
+kpplus=\e[+
+
+# Sequences below are used in OpenBSD wscons, also via remote login
+
+[terminal:vt220]
+bs=
+end=\e[8~
+f1=\e[11~
+f2=\e[12~
+f3=\e[13~
+f4=\e[14~
+f5=\e[15~
+f6=\e[17~
+f7=\e[18~
+f8=\e[19~
+f9=\e[20~
+f10=\e[21~
+f11=\e[23~
+f12=\e[24~
+home=\e[7~
+
+[terminal:wsvt25]
+bs=
+end=\e[8~
+f1=\e[11~
+f2=\e[12~
+f3=\e[13~
+f4=\e[14~
+f5=\e[15~
+f6=\e[17~
+f7=\e[18~
+f8=\e[19~
+f9=\e[20~
+f10=\e[21~
+f11=\e[23~
+f12=\e[24~
+home=\e[7~
+
+[terminal:wsvtg]
+bs=
+end=\e[8~
+f1=\e[11~
+f2=\e[12~
+f3=\e[13~
+f4=\e[14~
+f5=\e[15~
+f6=\e[17~
+f7=\e[18~
+f8=\e[19~
+f9=\e[20~
+f10=\e[21~
+f11=\e[23~
+f12=\e[24~
+home=\e[7~
+
+# Not a real vt100, but NetBSD wscons in vt220 mode with
+# $TERM set to vt100. Except F5-F9, this matches screen.
+
+[terminal:vt100]
+bs=
+complete=‰
+delete=\e[3~
+end=\e[8~
+f1=\e[11~
+f2=\e[12~
+f3=\e[13~
+f4=\e[14~
+f5=\e[15~
+f6=\e[17~
+f7=\e[18~
+f8=\e[19~
+f9=\e[20~
+f10=\e[21~
+f11=\e[23~
+f12=\e[24~
+home=\e[7~
+insert=\e[2~
+pgdn=\e[6~
+pgup=\e[5~
+
+# GNU screen
+[terminal:screen]
+bs=
+delete=\e[3~
+end=\e[8~
+f10=\e[20~
+f11=\e[23~
+f12=\e[24~
+f1=\e[11~
+f2=\e[12~
+f3=\e[13~
+f4=\e[14~
+f5=\e[15~
+# GNU screen says F6 is F5, but still...
+# maybe the bug will be fixed some time
+f6=\e[16~
+f7=\e[17~
+f8=\e[18~
+f9=\e[19~
+home=\e[7~
+insert=\e[2~
+pgdn=\e[6~
+pgup=\e[5~
+up=\eOA
+
+# PuTTY
+[terminal:ansi]
+bs=^h
+down=\eOB
+f1=\eOP
+f2=\eOQ
+f3=\eOR
+f4=\eOS
+left=\eOD
+right=\eOC
+up=\eOA
+
+# Windows 3.1 TELNET.EXE
+# (not really useful. The F-Keys, INS etc. don't work.)
+[terminal:ansi-nt]
+f1=\eOP
+f2=\eOQ
+f3=\eOR
+f4=\eOS
+bs=^h
+up=\e[A
+down=\e[B
+left=\e[D
+right=\e[C
+delete=
+
+# Microsoft Services for Unix 3.5 (native terminal, xterm see above)
+[terminal:interix]
+home=\e[H
+delete=
